app/src/main/java/com/codesoom/assignment/DemoHttpHandler.java

```diff
@@ -0,0 +1,223 @@
+
+                if (findTask.isPresent()) {
+                    Task task = findTask.get();
+                    responseBody = toJSON(task);
+                }
+
+            } catch (NumberFormatException e) {
+                code = 400;
+                responseBody = "Bad Request";
+            }
+
+        } else if (method.equals("DELETE") && path.contains("/tasks/")) {
+            try {
+                Long id = getId(path);
+                deleteTask(id);
+
+            } catch (NumberFormatException e) {
+                code = 400;
+                responseBody = "Bad Request";
+            }
+
+        } else {
+            code = 404;
+            responseBody = "Not Found";
+        }
+
+        resolveResponse(exchange, responseBody, code);
+    }
+
+    /**
+     * 클라이언트의 요청 body를 해결해 주는 메서드
+     */
+    private String resolveRequestBody(HttpExchange exchange) {
+        InputStream inputStream = exchange.getRequestBody();
+        InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
+        String json = new BufferedReader(inputStreamReader)
+                .lines()
+                .collect(Collectors.joining("\n")); //요청받은 body
+
+        if (!json.isBlank()) {
+            System.out.println(json);
+        }
+
+        return json;
+    }
+
+    //==편의 메서드==//
+
+    /**
+     * 클라이언트에서 받은 path에서 Task의 id 얻기
+     */
+    private Long getId(String path) {
+        String[] pathNames = path.split("/");
+        return Long.valueOf(pathNames[2]);
+    }
+
+    /**
+     * 클라이언트에 응답을 해결해 줄 메서드
+     */
+    private void resolveResponse(HttpExchange exchange, String responseBodyParam, int code) throws IOException {
+        byte[] responseBody = responseBodyParam.getBytes();
+        int responseBodyLength = responseBody.length;
+
+        exchange.sendResponseHeaders(code, responseBodyLength);
+
+        OutputStream outputStream = exchange.getResponseBody();
+        outputStream.write(responseBody);
+        outputStream.flush();
+        outputStream.close();
+    }
+
+    //==json 변환 메서드==//
+
+    /**
+     * Task를 JSON으로 포맷하기
+     */
+    private String toJSON(Task task) {
+        OutputStream outputStream = new ByteArrayOutputStream();
+
+        try {
+            objectMapper.writeValue(outputStream, task);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return outputStream.toString();
+    }
+
+    /**
+     * json을 Task로 포맷하기
+     */
+    private Task toTask(String json) throws JsonProcessingException {
+        return objectMapper.readValue(json, Task.class);
+    }
+
+    /**
+     * 할일(Task) 목록에서 주어인 아이디를 가진 할일(Task)을 찾기
+     */
+    private Optional<Task> findTaskById(Long id) {
+        Optional<Task> result = Optional.empty();
+
+        for (Task task : tasks) {
+            if (task.getId().equals(id)) {
+                result = Optional.of(task);
+                break;
+            }
+        }
+
+        return result;
+    }
```

Comment:
```suggestion
    private Optional<Task> findTaskById(Long id) {
        for (Task task : tasks) {
            if (task.getId().equals(id)) {
                return Optional.of(task);
            }
        }

        return Optional.empty();
    }
```

break문 없이 이렇게 하는 것은 어떨까요?

Reply:
f399ae3 break문을 제거하고 return문을 활용했습니다.

코드가 좀 더 간결해짐을 느꼈습니다.